File: uw-count-back-end/src/main/java/org/uwcount/dto/Invoice.java
package org.uwcount.dto;

import lombok.*;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Invoice {
    private Integer id;
    private Integer customerId;
    private String invoiceNo;
    private LocalDate date;
    private LocalDate dueDate;
    private Double subTotal;
    private Double discount;
    private Double tax;
    private Double total;
    private List<InvoiceLine> invoiceLines;
}

File: uw-count-front-end/src/app/models/invoice.ts
import { InvoiceLine } from "./invoice-line";

export interface Invoice {
  id:           number|null;
  customerId:   number;
  invoiceNo:    string;
  date:         string;    // "YYYY-MM-DD"
  dueDate:      string;
  subTotal:     number;
  discount:     number;
  tax:          number;
  total:        number;
  invoiceLines: InvoiceLine[];
}